framework/src/systems/NonlinearSystemBase.C

```diff
@@ -3311,33 +3317,11 @@ NonlinearSystemBase::setPreviousNewtonSolution(const NumericVector<Number> & sol
 void
 NonlinearSystemBase::mortarConstraints()
 {
-  const auto & undisplaced_mortar_interfaces = _fe_problem.getMortarInterfaces(/*displaced=*/false);
-
-  for (const auto & mortar_interface : undisplaced_mortar_interfaces)
-  {
-    auto it = _undisplaced_mortar_functors.find(mortar_interface.first);
+  for (auto & map_pr : _undisplaced_mortar_functors)
+    map_pr.second();
 
-    mooseAssert(it != _undisplaced_mortar_functors.end(),
-                "No ComputeMortarFunctor exists for the specified primary-secondary boundary "
-                "pair, primary "
-                    << mortar_interface.first.first << " and secondary "
-                    << mortar_interface.first.second);
-    it->second();
-  }
-
-  const auto & displaced_mortar_interfaces = _fe_problem.getMortarInterfaces(/*displaced=*/true);
-
-  for (const auto & mortar_interface : displaced_mortar_interfaces)
-  {
-    auto it = _displaced_mortar_functors.find(mortar_interface.first);
-
-    mooseAssert(it != _displaced_mortar_functors.end(),
-                "No ComputeMortarFunctor exists for the specified primary-secondary boundary "
-                "pair, primary "
-                    << mortar_interface.first.first << " and secondary "
-                    << mortar_interface.first.second);
-    it->second();
-  }
+  for (auto & map_pr : _displaced_mortar_functors)
+    map_pr.second();
```

Comment:
Now, we don't add the check that each functor must have a particular primary-secondary pair. I can't see if the added code is taking care of possible error.

Reply:
What possible error? We should only be looping over primary-secondary pairs that were requested by the user through creation of mortar aux kernels and mortar constraints. So there should be no possibility of an error here

Reply:
Ok. Was referring to the removal of the assertion:

``` mooseAssert(it != _displaced_mortar_functors.end(),
                "No ComputeMortarFunctor exists for the specified primary-secondary boundary "
                "pair, primary "
                    << mortar_interface.first.first << " and secondary "
                    << mortar_interface.first.second);